kraken-java-sdk/kraken-java-sdk-controller/src/main/java/com/consoleconnect/kraken/operator/controller/service/push/ApiActivityPushService.java

```diff
@@ -0,0 +1,132 @@
+package com.consoleconnect.kraken.operator.controller.service.push;
+
+import static com.consoleconnect.kraken.operator.core.toolkit.JsonToolkit.fromJson;
+import static com.consoleconnect.kraken.operator.core.toolkit.JsonToolkit.toJson;
+
+import com.consoleconnect.kraken.operator.controller.dto.push.ApiRequestActivityPushResult;
+import com.consoleconnect.kraken.operator.controller.dto.push.CreatePushApiActivityRequest;
+import com.consoleconnect.kraken.operator.controller.dto.push.PushApiActivityLogEnabled;
+import com.consoleconnect.kraken.operator.controller.dto.push.PushApiActivityLogHistory;
+import com.consoleconnect.kraken.operator.controller.model.Environment;
+import com.consoleconnect.kraken.operator.controller.service.EnvironmentService;
+import com.consoleconnect.kraken.operator.core.entity.MgmtEventEntity;
+import com.consoleconnect.kraken.operator.core.enums.EventStatusType;
+import com.consoleconnect.kraken.operator.core.enums.MgmtEventType;
+import com.consoleconnect.kraken.operator.core.exception.KrakenException;
+import com.consoleconnect.kraken.operator.core.repo.MgmtEventRepository;
+import com.consoleconnect.kraken.operator.core.request.PushLogSearchRequest;
+import com.consoleconnect.kraken.operator.core.toolkit.Paging;
+import com.consoleconnect.kraken.operator.core.toolkit.PagingHelper;
+import jakarta.persistence.criteria.Predicate;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class ApiActivityPushService {
+
+  public static final String CREATED_AT = "createdAt";
+  public static final String EVENT_TYPE = "eventType";
+
+  private final MgmtEventRepository mgmtEventRepository;
+  private final EnvironmentService environmentService;
+
+  @Value("${app.features.push-activity-log-external.enabled}")
+  private boolean pushActivityLogExternalEnabled;
+
+  public ApiRequestActivityPushResult createPushApiActivityLogInfo(
+      CreatePushApiActivityRequest request, String userId) {
```

Comment:
If the feature disables,  the  method  createPushApiActivityLogInfo should  not  execute the  request and tell user info about  feature is disabled